Automattic/wp-calypso
# Site Profiler: Show a different title for the hosting section according the provider

Fixes https://github.com/Automattic/dotcom-forge/issues/7315

## Proposed Changes

* Isolate the HostingSection on its own component
* Show a different message for WP.com hosted sites
* Generalize translateOptions on an util function

## Why are these changes being made?
To match the new layout for Site Profiler v2

## Testing Instructions
* Go to `/site-profiler/:url` with a non-WP.com site. Ex: `/site-profiler/example.com`
* Check if the title is similar to the picture below
* Go to `/site-profiler/:url` with a WP.com site. Ex: `/site-profiler/wordpress.com`
* Check if the title is similar to the picture below

| WP.com  | non-WP.com |
| ------------- | ------------- |
|![CleanShot 2024-05-22 at 12 14 07@2x](https://github.com/Automattic/wp-calypso/assets/5039531/0b91bdbc-7f54-4f81-a09b-b94018acd3fd)|![CleanShot 2024-05-22 at 12 14 17@2x](https://github.com/Automattic/wp-calypso/assets/5039531/879434a9-abb7-4621-aaab-de8fbfe09af0)|


## Pre-merge Checklist
